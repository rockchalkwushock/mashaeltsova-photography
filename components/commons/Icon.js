import PropTypes from 'prop-types'
import FontAwesome from 'react-fontawesome'

// Get current year from browser.
const year = new Date().getFullYear()
// Slick thinking!
const yearToDisplay = year === 2017 ? year : `2017 - ${year}`

// NOTE: She wants this left in English!
const Icon = ({ icon, size, url }) => {
  if (icon === 'copyright') {
    return (
      <a href={url}>
        <span>{`Designed by RCWS Development ${yearToDisplay}`}</span>
        <style jsx>{`
          @media (max-width: 321px) {
            span {
              font-size: 0.95em;
            }
          }
        `}</style>
      </a>
    )
  }
  return (
    <a href={url}>
      <FontAwesome name={icon} size={size} />
    </a>
  )
}

Icon.propTypes = {
  icon: PropTypes.string.isRequired,
  size: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}

export default Icon
