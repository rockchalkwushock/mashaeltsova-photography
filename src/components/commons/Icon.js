import React from 'react'
import styled from 'styled-components'
import fontawesome from '@fortawesome/fontawesome'
import FontAwesomeIcon from '@fortawesome/react-fontawesome'
import faBars from '@fortawesome/fontawesome-free-solid/faBars'
import faEnvelope from '@fortawesome/fontawesome-free-solid/faEnvelope'
import faFacebook from '@fortawesome/fontawesome-free-brands/faFacebook'
import faInstagram from '@fortawesome/fontawesome-free-brands/faInstagram'
import faVk from '@fortawesome/fontawesome-free-brands/faVk'

fontawesome.library.add(faBars, faFacebook, faEnvelope, faInstagram, faVk)

const UnstyledIcon = props => {
  // Removes error: 'Received "menu": `true` for non-boolean from fontawesome
  const rest = { ...props, menu: null }
  return <FontAwesomeIcon {...rest} />
}

export const StyledIcon = styled(UnstyledIcon)`
  margin-left: ${({ menu }) => (menu ? '10px' : 'unset')}
  margin-top: ${({ menu }) => (menu ? '10px' : 'unset')}
`

const Icon = props => <StyledIcon {...props} />

export default Icon
