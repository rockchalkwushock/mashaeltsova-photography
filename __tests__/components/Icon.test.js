import React from 'react'
import { render } from 'enzyme'

import Icon from '../../components/commons/Icon'

describe('Component: Icon', () => {
  test('Renders without exploding as copyright icon', () => {
    const tree = render(
      <Icon icon="copyright" size="lg" url="https://rcws-development.com" />
    )
    expect(tree).toMatchSnapshot()
  })
  test('Renders without exploding as social icon', () => {
    const tree = render(
      <Icon icon="instagram" size="2x" url="https://instagram.com" />
    )
    expect(tree).toMatchSnapshot()
  })
})
