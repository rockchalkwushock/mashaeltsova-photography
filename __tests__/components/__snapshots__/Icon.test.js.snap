// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Icon Renders without exploding as copyright icon 1`] = `
<a
  class="jsx-1299221301"
  href="https://rcws-development.com"
>
  <span
    class="jsx-1299221301"
  >
    Designed by RCWS Development 2017
  </span>
</a>
`;

exports[`Component: Icon Renders without exploding as social icon 1`] = `
<a
  href="https://instagram.com"
>
  <span
    aria-hidden="true"
    class="fa fa-instagram fa-2x"
  />
</a>
`;
